docs/user_guide/secure_payloads.rst

```diff
@@ -0,0 +1,96 @@
+Secure Payloads
+================
+
+.. warning::
+    This page is still under development and not complete. It will be so until
+    this warning is removed.
+
+Trusted payloads offers the ability to securely provision secrets to an enrolled node.
+
+The payload itself is sent to via the Keylime Tenant to the verifier and only when a node
+has passed its enrollment criteria (including any `tpm_policy` or IMA whitelist), will the 
+payload be passed to the agent for execution.
+
+The follow flow will occur during execution of a Trusted payload:
+
+* The tenant will first set up a folder containing the files to be copied and the scripts to be executed by the agent
+* The tenant will generate a new CA (if one hasn't been generated yet) and then generate a new cert/key combo for the agent to be bootstrapped
+* The tenant will include the payload files with the `--include` option
+* Tenant and verifier will bootstrap the agent and derive a key as normal
+* Once the bootstrap key has been derived and tpm_policy and IMA verification is passed, keylime will decrypt and extract the zipped payload that includes both the cert/key and files situated in the payload folder.
+* Then keylime will run the provided autorun.sh script.
+
+The scripts work by leveraging two features of keylime: automatic certificate generation/delivery
+and the ability run scripts provided in the payload after successful bootstrapping.
+
+
+For the following example, we will provision some SSH keys onto the agent.
+
+1. Create a directory to host the files and `autorun.sh` script. For this example, we will use `payload`
+2. Create an `autorun.sh` script
+
+```
+#!/bin/bash
+
+mkdir -p /root/.ssh/
+cp payload_id_rsa* /root/.ssh/
+chmod 600 /root/.ssh/payload_id_rsa*
+```
+
+3. Copy in the files you wish to provision within the payload to the `payload` directory.
+
+```
+$ ls payload/
+payload_id_rsa.pub
+payload_id_rsa
+```
+
+Send the files using the Keylime Tenant
+
+```
+keylime_tenant -t <agent-ip> --cert myca --include ssh_revoke
```

Comment:
Is `--include` using `payload/` here, or `ssh_revoke`?  Where is `ssh_revoke` defined?